akka-docs-dev/rst/java/stream-graphs.rst

```diff
@@ -0,0 +1,212 @@
+from design discussions, or illustrations in protocol specifications) to and from code simpler. In this section we'll
+dive into the multiple ways of constructing and re-using graphs, as well as explain common pitfalls and how to avoid them.
+
+Graphs are needed whenever you want to perform any kind of fan-in ("multiple inputs") or fan-out ("multiple outputs") operations.
+Considering linear Flows to be like roads, we can picture graph operations as junctions: multiple flows being connected at a single point.
+Some graph operations which are common enough and fit the linear style of Flows, such as ``concat`` (which concatenates two
+streams, such that the second one is consumed after the first one has completed), may have shorthand methods defined on
+:class:`Flow` or :class:`Source` themselves, however you should keep in mind that those are also implemented as graph junctions.
+
+.. _flow-graph-java:
+
+Constructing Flow Graphs
+------------------------
+Flow graphs are built from simple Flows which serve as the linear connections within the graphs as well as junctions
+which serve as fan-in and fan-out points for Flows. Thanks to the junctions having meaningful types based on their behaviour
+and making them explicit elements these elements should be rather straightforward to use.
+
+Akka Streams currently provide these junctions:
+
+* **Fan-out**
+
+ - ``Broadcast<T>`` – (1 input, n outputs) signals each output given an input signal,
+ - ``Balance<T>`` – (1 input => n outputs), signals one of its output ports given an input signal,
+ - ``UnZip<A,B>`` – (1 input => 2 outputs), which is a specialized element which is able to split a stream of ``(A,B)`` tuples into two streams one type ``A`` and one of type ``B``,
+ - ``FlexiRoute<In>`` – (1 input, n outputs), which enables writing custom fan out elements using a simple DSL,
+
+* **Fan-in**
+
+ - ``Merge<In>`` – (n inputs , 1 output), picks signals randomly from inputs pushing them one by one to its output,
+ - ``MergePreferred<In>`` – like :class:`Merge` but if elements are available on ``preferred`` port, it picks from it, otherwise randomly from ``others``,
+ - ``ZipWith<A,B,...,Out>`` – (n inputs (defined upfront), 1 output), which takes a function of n inputs that, given all inputs are signalled, transforms and emits 1 output,
+ - ``Zip<A,B>`` – (2 inputs, 1 output), which is a :class:`ZipWith` specialised to zipping input streams of ``A`` and ``B`` into an ``(A,B)`` tuple stream,
+ - ``Concat<A>`` – (2 inputs, 1 output), which enables to concatenate streams (first consume one, then the second one), thus the order of which stream is ``first`` and which ``second`` matters,
+ - ``FlexiMerge<Out>`` – (n inputs, 1 output), which enables writing custom fan out elements using a simple DSL.
+
+One of the goals of the FlowGraph DSL is to look similar to how one would draw a graph on a whiteboard, so that it is
```

Comment:
Just had a thought that that's not really valid for the javadsl - it's just `addEdge`, not very visual..